backup: TestOnlineRestoreDistFlowSplitScatter failed

backup.TestOnlineRestoreDistFlowSplitScatter [failed](https://mesolite.cluster.engflow.com/invocations/default/b0b57007-f398-4245-824d-5b9028f6afaa?testReportRun=6&testReportShard=47&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [7e489320238b6feb3fcf94afa5727b7dff7cf51f](https://github.com/cockroachdb/cockroach/commits/7e489320238b6feb3fcf94afa5727b7dff7cf51f):


```
=== RUN   TestOnlineRestoreDistFlowSplitScatter
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestOnlineRestoreDistFlowSplitScatter4211208810
    test_log_scope.go:82: use -show-logs to present logs inline
    restore_online_distflow_test.go:188: -- test log scope end --
test logs left over in: outputs.zip/logTestOnlineRestoreDistFlowSplitScatter4211208810
--- FAIL: TestOnlineRestoreDistFlowSplitScatter (34.74s)
=== RUN   TestOnlineRestoreDistFlowSplitScatter/goroutine-path
    restore_online_distflow_test.go:174: error executing query="CANCEL JOB 1164800841453043714" args=[]: pq: job 1164800841453043714: job with state pause-requested cannot be requested to be canceled
--- FAIL: TestOnlineRestoreDistFlowSplitScatter/goroutine-path (0.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=47</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreDistFlowSplitScatter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62569

backup.TestOnlineRestoreDistFlowSplitScatter [failed](https://mesolite.cluster.engflow.com/invocations/default/181a8809-3f58-4b1d-a018-802cc147ff5e?testReportRun=7&testReportShard=47&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [f9bcbc4fdf0ba5236f6312c00becf303c29f98f9](https://github.com/cockroachdb/cockroach/commits/f9bcbc4fdf0ba5236f6312c00becf303c29f98f9):


```
=== RUN   TestOnlineRestoreDistFlowSplitScatter
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestOnlineRestoreDistFlowSplitScatter3844608995
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:635: DRPC is enabled (via testserver factory defaults)
    restore_online_distflow_test.go:188: -- test log scope end --
test logs left over in: outputs.zip/logTestOnlineRestoreDistFlowSplitScatter3844608995
--- FAIL: TestOnlineRestoreDistFlowSplitScatter (33.59s)
=== RUN   TestOnlineRestoreDistFlowSplitScatter/goroutine-path
    restore_online_distflow_test.go:174: error executing query="CANCEL JOB 1166492000862502914" args=[]: pq: job 1166492000862502914: job with state pause-requested cannot be requested to be canceled
--- FAIL: TestOnlineRestoreDistFlowSplitScatter/goroutine-path (0.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=47</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreDistFlowSplitScatter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/restore_online_distflow_test.go
+++ b/pkg/backup/restore_online_distflow_test.go
@@ -15,11 +15,13 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/backup/backuppb"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/cloud/nodelocal"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/sql"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/jobutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
@@ -113,6 +115,7 @@ func TestOnlineRestoreDistFlowSplitScatter(t *testing.T) {
 					},
 				},
 			},
+			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
 		},
 	})
 
@@ -171,7 +174,10 @@ func TestOnlineRestoreDistFlowSplitScatter(t *testing.T) {
 		var downloadJobID jobspb.JobID
 		runner.QueryRow(t, latestDownloadJobIDQuery).Scan(&downloadJobID)
 		runner.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = ''")
-		runner.Exec(t, fmt.Sprintf("CANCEL JOB %d", downloadJobID))
+		testutils.SucceedsSoon(t, func() error {
+			_, err := sqlDB.Exec(fmt.Sprintf("CANCEL JOB %d", downloadJobID))
+			return err
+		})
 		jobutils.WaitForJobToCancel(t, runner, downloadJobID)
 		runner.Exec(t, "SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_download'")
 	}

